pkg/globalnet/controllers/ipam/ipam.go

```diff
@@ -0,0 +1,308 @@
+package ipam
+
+import (
+	"fmt"
+	"sync"
+	"time"
+
+	k8sv1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/retry"
+	"k8s.io/client-go/util/workqueue"
+	"k8s.io/klog"
+)
+
+func NewController(spec *SubmarinerIpamControllerSpecification, config *InformerConfigStruct) (*Controller, error) {
+	exclusionMap := make(map[string]bool)
+	for _, v := range spec.ExcludeNS {
+		exclusionMap[v] = true
+	}
+	pool, err := NewIpPool(spec.GlobalCIDR)
+	if err != nil {
+		return nil, err
+	}
+
+	ipamController := &Controller{
+		kubeClientSet:    config.KubeClientSet,
+		serviceWorkqueue: workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), "Services"),
+		servicesSynced:   config.ServiceInformer.Informer().HasSynced,
+		podWorkqueue:     workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), "Pods"),
+		podsSynced:       config.PodInformer.Informer().HasSynced,
+
+		excludeNamespaces: exclusionMap,
+		pool:              pool,
+	}
+	klog.Info("Setting up event handlers")
+	config.ServiceInformer.Informer().AddEventHandlerWithResyncPeriod(cache.ResourceEventHandlerFuncs{
+		AddFunc: func(obj interface{}) {
+			ipamController.enqueueObject(obj, ipamController.serviceWorkqueue)
+		},
+		UpdateFunc: func(old, new interface{}) {
```

Comment:
`new` is a language keyword. I would avoid using that name. They usually use `newObj` and `oldObj`

There are more places where this is present. Only highlighting it here.

Reply:
will do. old naming habits.